### PR TITLE
commit

### PR DESCRIPTION
**Pull request by josoriom**

### CODE REVIEW

**Comment by lpatiny** [comment]
@targos Please could you review this code. The results are exactly the same as with R so we are quite confident the algorithm is correct.
I think the package.json is not clean and their are many mistakes
Concerning the call the the method (arguments of the function) I'm don't know.
Same is true for the algorithm it self that uses really a lot of maps / reduce.

File: .yo-rc.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (targos) [inline .yo-rc.json]: please remove this file

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2019 cheminfo
```
Comment (targos) [inline LICENSE]:
```suggestion
Copyright (c) 2019 mljs
```

File: README.md [hunk 3]
```diff
@@ -1 +1,44 @@
+## API
```
Comment (targos) [inline README.md]:
```suggestion
# pcr

Principal Component Regression.

## API
```

File: README.md [hunk 4]
```diff
@@ -1 +1,44 @@
...
 
+### new PCR(x, y, intercept, weight)
```
Comment (targos) [inline README.md]: Please implement this as an options object. For example: `new PCR(x, y, { intercept: true, weight: 0.5 })`.

File: README.md [hunk 5]
```diff
@@ -1 +1,44 @@
...
+
+**Arguments**
+
+* `x`: Matrix containing the inputs.
+* `y`: Matrix containing the outputs.
+* `intercept`: boolean indicating if intercept terms should be computed.
+* `weight`: number (0 - 100): It refers to the weight that components must sum with each other (in percent) to perform the regression. When this is 100%, pcr perform a multiple linear regression.
```
Comment (targos) [inline README.md]: Percentage should be given in range 0-1

**Comment by targos** [comment]
Do you have a reference for the algorithm?

**Comment by josoriom** [comment]
Sorry I do not have a reference.  I read a lot of books and info in the
network. But I used a example of (Miller, J. N., & Miller, J. C. (2010).
Statistics and Chemometrics for Analytical Chemistry. pp(238-243)) to
validate my data.

**Pull request merged**